test/test_cuda.py

```diff
@@ -16,6 +17,15 @@
     TestCase = object  # noqa: F811
     HAS_CUDA = False
 
+HAS_MAGMA = HAS_CUDA
+if HAS_CUDA:
+    try:
+        x = torch.ones(1, 1).cuda()
+        x.symeig()
+    except RuntimeError as err:
+        if re.search("No CUDA implementation of '\w+'. Install MAGMA and", str(err)) is not None:
```

Comment:
Let's try to avoid this sort of detection. There's `torch.cuda.has_magma`. It's a little weird in that it's only set once CUDA is initialized (we should fix that).

Reply:
my bad, i didn't know torch.cuda.has_magma existed, so I told Richard there's no coherent way to do magma detection at the moment